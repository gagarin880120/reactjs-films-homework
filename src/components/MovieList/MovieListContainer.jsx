import React, { useEffect, useMemo, createRef, useState } from 'react';
import MovieList from './MovieList';
import styles from './MovieList.module.scss';
import { useSelector, useDispatch } from 'react-redux';
import { searchResultsSelector, genresSelector } from '../../redux/selectors';
import { getGenres } from '../../redux/actions';

export default function MovieListContainer() {
  const results = useSelector(searchResultsSelector);
  const genres = useSelector(genresSelector);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getGenres());
  }, [])

  return (
    <div>
      <MovieList
        results={results}
        genres={genres}
      />
    </div>
  )
}
