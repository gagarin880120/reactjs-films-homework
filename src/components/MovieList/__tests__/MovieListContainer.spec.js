import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import MovieListContainer from '../MovieListContainer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

test('MovieListContainer component should render without crashing', () => {
  const mockStore = configureStore();
  const store = mockStore({});
  const renderer = new ShallowRenderer();
  renderer.render(<Provider store={store}><MovieListContainer /></Provider>);
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
