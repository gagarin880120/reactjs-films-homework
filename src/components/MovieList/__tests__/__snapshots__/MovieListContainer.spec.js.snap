// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieListContainer component should render without crashing 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <MovieListContainer />
</Context.Provider>
`;
